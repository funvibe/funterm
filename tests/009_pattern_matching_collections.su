lua.array = [[8, "a", 8], ["a", "a", 1], [], [1]]
for i = 0,3 do
  match lua.array[i] {
    []          -> lua.print("empty list"),
    [_, _, 1]   -> lua.print("1"),
    [_, "a", _] -> lua.print("a"),
    _           -> lua.print("No match")
  }
end

lua.dict = [{"city": "Moscow"}, {"region": 77}, {}, {"age": 878}]
for i = 0,3 do
  match lua.dict[i] {
    {}             -> lua.print("empty dict"),
    {"city": city} -> lua.print(city),
    {_: 77}        -> lua.print("region"),
    _              -> lua.print("No match")
  }
end