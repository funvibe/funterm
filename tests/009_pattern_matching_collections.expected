a
1
empty list
No match
Moscow
region
empty dict
No match